src/main/java/dev/dokan/dokan_java/conv/MaskValueSet.java

```diff
@@ -0,0 +1,56 @@
+package dev.dokan.dokan_java.conv;
+
+import dev.dokan.dokan_java.constants.dokany.MountOption;
+import dev.dokan.dokan_java.constants.microsoft.FileSystemFlag;
+
+import java.util.Set;
+
+/**
+ * Used to store multiple {@link MaskValueEnum} values such as {@link FileSystemFlag} and {@link MountOption}.
+ *
+ * @param <T> Type of {@link EnumInteger}
+ */
+public interface MaskValueSet<T extends Enum<T> & MaskValueEnum> extends Set<T> {
+
+    static <T extends Enum<T> & MaskValueEnum> MaskValueSet<T> emptySet(Class<T> clazz) {
+        return new MaskValueSetImpl<>(clazz);
+    }
+
+    @SafeVarargs
+    static <T extends Enum<T> & MaskValueEnum> MaskValueSet<T> of(T first, T... others) {
+        return new MaskValueSetImpl<>(first, others);
+    }
+
+    /**
+     * Creates a set of MaskValueEnums which corresponds to the bit flag given as an 32bit integer.
+     * <p>
+     * The type of the set is the enum class of the input array
+     *
+     * @param intValue the integer value of the combined bitflag
+     * @param allEnumValues all possible values of this MaskValueEnum
+     * @param <T> enum type of the array implementing the MaskValueEnum interface
+     * @return a set of MaskValueEnum values whose mask were set in the intValue
+     */
+    static <T extends Enum<T> & MaskValueEnum> MaskValueSet<T> getSetFromInt(final int intValue, final T[] allEnumValues) {
+        MaskValueSet<T> elements = new MaskValueSetImpl<>(allEnumValues[0].getDeclaringClass());
+        int remainingValues = intValue;
+        for (T current : allEnumValues) {
+            int mask = current.intValue();
+
+            if ((remainingValues & mask) == mask) {
+                elements.add(current);
+                remainingValues -= mask;
+            }
+        }
+        return elements;
+    }
+
+    static <T extends Enum<T> & MaskValueEnum> MaskValueSet<T> getSetFromInt(final int intValue, final Class<T> type) {
```

Comment:
also here one should think about the naming

Reply:
Resolved with commit e042b3d44d5b944f198f2cd3a2dca89419f56e8f / 3114b10f633ea2c173307afff6119c208ba7d791